kong/clustering/rpc/manager.lua

```diff
@@ -232,61 +347,57 @@ end
 
 -- handle incoming client connections
 function _M:handle_websocket()
-  local kong_version = ngx_var.http_x_kong_version
   local node_id = ngx_var.http_x_kong_node_id
   local rpc_protocol = ngx_var.http_sec_websocket_protocol
-  local content_encoding = ngx_var.http_content_encoding
-  local rpc_capabilities = ngx_var.http_x_kong_rpc_capabilities
-
-  if not kong_version then
-    ngx_log(ngx_ERR, "[rpc] client did not provide version number")
-    return ngx_exit(ngx.HTTP_CLOSE)
-  end
 
   if not node_id then
     ngx_log(ngx_ERR, "[rpc] client did not provide node ID")
     return ngx_exit(ngx.HTTP_CLOSE)
   end
 
-  if content_encoding ~= "x-snappy-framed" then
-    ngx_log(ngx_ERR, "[rpc] client does use Snappy compressed frames")
-    return ngx_exit(ngx.HTTP_CLOSE)
+  local meta_v1_supported
+  local protocols = string_tools.split(rpc_protocol, ",")
+
+  -- choice a proper protocol
+  for _, v in ipairs(protocols) do
+    -- now we only support kong.meta.v1
+    if RPC_MATA_V1 == string_tools.strip(v) then
+      meta_v1_supported = true
+      break
+    end
   end
```

Comment:
Do we have a `table.find` or equivalent?

Reply:
No idea, sorry.